hooks/useGas.ts

```diff
@@ -46,10 +71,27 @@ const useGas = () => {
 		}
 	}, []);
 
+	const gasConfig = useMemo(() => {
+		const maxFeePerGasValue = isCustomGasPrice ? gasPriceWei : maxFeePerGas;
+		const l1GasConfig = { maxPriorityFeePerGas, maxFeePerGas: maxFeePerGasValue };
+		const l2GasConfig = { gasPrice: gasPriceWei };
+
+		return isL2 ? l2GasConfig : l1GasConfig;
+	}, [gasPriceWei, isCustomGasPrice, isL2, maxFeePerGas, maxPriorityFeePerGas]);
```

Comment:
this object here is the main difference in this pr, it is used across the components to pass the right gas configuration to the transactions.